### ISSUE TITLE
Read validator graffiti from file

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#2074]
## Description

Some users have been requesting some funky graffiti requirements (e.g., drawing stuff on beaconcha.in walls, etc). I think we can satisfy practically any scenario by just reading the graffiti from a file.

## Steps to resolve

- Add a `--graffiti-file` flag to BN, VC.
- Try to read from the file each time we produce/sign a block (not just at startup).
- If the file read fails or the value is too long, default back to `--graffitti`.

**Comment by michaelsproul** [comment]
I guess this would supersede https://github.com/sigp/lighthouse/pull/2044

I don't think we'd need any option to set it per validator in that case?

**Comment by GregTheGreek** [comment]
I (Lodestar) will be working on a service to update the file in a nice seamless way :)

**Comment by pawanjay176** [comment]
> Do you know what the intended usecase for per-validator graffiti was

I assumed it was to sorta improve validator privacy. i.e. if 10 vals running on a single vc have the same graffiti vs each having different graffiti. 

If the graffiti-file will keep getting updated for the graffiti wall, I don't think we need separate graffiti's per validator. But I might be missing some other use case here.

**Comment by GregTheGreek** [comment]
ACtually that would be optimal imo. It would be nice to be able to delegate which validator pulls which graffiti..
Could the config resemble something like:
```
default: hi # Represents the fallback incase of an error
v1: foo
v2: bar
...
```

**Comment by realbigsean** [comment]
I liked the idea of being able to manage graffiti via the VC API, which was made easy by adding graffiti to the `validator_definitions.yml` like in #2044. Is the reason to add a separate file for graffiti so we don't have to reload the whole `validator_definitions.yml` file? Or so that we don't promote editing that file frequently or automatically? 

Either way I'm definitely in favor of enabling per-validator graffiti

**Comment by pawanjay176** [comment]
> Is the reason to add a separate file for graffiti so we don't have to reload the whole `validator_definitions.yml`. Or so that we don't promote editing that file frequently or automatically?

I think both are correct. 

I too liked having the graffiti in the `validator_definitions.yml` as it was simple for the user (user just has to enter an extra graffiti entry for required validators).

With the separate file, the user would have to manually copy pubkey entries from `validator_definitions.yml` to the file in the beginning. Would the service also be doing this initialisation @GregTheGreek ?

**Comment by remyroy** [comment]
This is related to #2031 and #1944.

I like having it both in `validator_definitions.yml` and being able to manage the graffiti via the VC API.

**Comment by ixaeon** [comment]
Since the graffiti-file isn't critical to system function, it could be its own file separate from `validator_definitions.yml` which would be less specific and make sense for the beacon to use.

The user gets the benefit of being able to simply replace or edit a single file that won't interfere with system operation or cause anything to need reloading.

**Comment by pawanjay176** [comment]
@ixaeon added support for a separate graffiti file in #2044

**Comment by pawanjay176** [comment]
Fixed in #2044
